app/main.py

```diff
@@ -2,0 +1,35 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: "str", health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __str__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    @staticmethod
+    def die(animal: "Animal") -> None:
+        if animal in Animal.alive:
+            Animal.alive.remove(animal)
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
You have correctly used the idea of changing a boolean value in one line. This is a good practice and improves the efficiency of your code. Well done!